src/net_processing.cpp

```diff
@@ -867,6 +907,13 @@ void PeerLogicValidation::ReattemptInitialBroadcast(CScheduler& scheduler) const
 }
 
 void PeerLogicValidation::FinalizeNode(NodeId nodeid, bool& fUpdateConnectionTime) {
+    int misbehavior;
+    {
+        PeerRef peer = GetPeerRef(nodeid);
```

Comment:
36ac41a should we add an `if (peer == nullptr) return false;` check here as well

Reply:
I've added an assert, like for CNodeState. The Peer object should always exist when we try to remove the peer.